pkg/collector/logical_disk/logical_disk.go

```diff
@@ -271,118 +290,144 @@ func (c *collector) collect(ctx *types.ScrapeContext, ch chan<- prometheus.Metri
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.AvgReadQueue,
+					prometheus.GaugeValue,
+					volume.AvgDiskReadQueueLength*perflib.TicksToSecondScaleFactor,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.AvgWriteQueue,
+					prometheus.GaugeValue,
+					volume.AvgDiskWriteQueueLength*perflib.TicksToSecondScaleFactor,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.ReadBytesTotal,
+					prometheus.CounterValue,
+					volume.DiskReadBytesPerSec,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.ReadsTotal,
+					prometheus.CounterValue,
+					volume.DiskReadsPerSec,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.WriteBytesTotal,
+					prometheus.CounterValue,
+					volume.DiskWriteBytesPerSec,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.WritesTotal,
+					prometheus.CounterValue,
+					volume.DiskWritesPerSec,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.ReadTime,
+					prometheus.CounterValue,
+					volume.PercentDiskReadTime,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.WriteTime,
+					prometheus.CounterValue,
+					volume.PercentDiskWriteTime,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.FreeSpace,
+					prometheus.GaugeValue,
+					volume.PercentFreeSpace_Base*1024*1024,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.TotalSpace,
+					prometheus.GaugeValue,
+					volume.PercentFreeSpace*1024*1024,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.IdleTime,
+					prometheus.CounterValue,
+					volume.PercentIdleTime,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.SplitIOs,
+					prometheus.CounterValue,
+					volume.SplitIOPerSec,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.ReadLatency,
+					prometheus.CounterValue,
+					volume.AvgDiskSecPerRead*perflib.TicksToSecondScaleFactor,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.WriteLatency,
+					prometheus.CounterValue,
+					volume.AvgDiskSecPerWrite*perflib.TicksToSecondScaleFactor,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+				ch <- prometheus.MustNewConstMetric(
+					c.ReadWriteLatency,
+					prometheus.CounterValue,
+					volume.AvgDiskSecPerTransfer*perflib.TicksToSecondScaleFactor,
+					volume.Name,
+					logicalDisk.VolumeName,
+				)
+
+			} else {
+				continue
```

Comment:
this this an error path? If yes, we should throw any error here instead silent continue

Reply:
Hello! 
No, this is just processing the loop condition. In this loop we are looking for a match of volume(for example, C: == C: ). And if we find a match, then we write down the metrics, if a match is not found at the current iteration, we continue to search